pkg/handler/validate.go

```diff
@@ -47,3 +48,22 @@ func validatePutDistributions(params distribution.PutDistributionsParams) *Error
 
 	return nil
 }
+
+func validateDeleteVariant(params variant.DeleteVariantParams) *Error {
+	f := &entity.Flag{}
+	err := entity.NewFlagQuerySet(repo.GetDB()).IDEq(uint(params.FlagID)).One(f)
+	if err != nil {
+		return NewError(400, "error finding flagID %v. reason %s", params.FlagID, err)
+	}
+	f.Preload(repo.GetDB())
+
+	for _, s := range f.Segments {
+		for _, d := range s.Distributions {
+			if d.VariantID == util.SafeUint(params.VariantID) && d.Percent != uint(0) {
+				return NewError(400, "error deleting variant %v. distribution %v still has non-zero distribution %v", params.VariantID, d.ID, d.Percent)
```

Comment:
Note, we cannot delete a variant if it's rolling out to some non-zero distribution

Reply:
oo ok, yea, i can put that check on the frontend too

Reply:
yup, actually I need to remove all the distribution that has the references to it

Reply:
ohh, so do you want it so when they delete a variant, it deletes all referenced distributions? or just not allow that until they delete the distribution?

Reply:
going to delete the referenced distribution if percent is `0`, otherwise, return 400

Reply:
oo ok